test/middleware.spec.ts

```diff
@@ -53,8 +66,95 @@ describe("middleware test", () => {
   beforeAll(() => {
     listen(TEST_PORT, m);
   });
+
+  describe("With skipSignatureVerification functionality", () => {
```

Comment:
Aren't these tests run in parallel? 

To avoid flaky tests, can we design the tests so they don't share state?